logic/src/commonMain/kotlin/com/wire/kalium/logic/di/AuthenticatedDataSourceSetProvider.kt

```diff
@@ -0,0 +1,27 @@
+package com.wire.kalium.logic.di
+
+import com.wire.kalium.logic.AuthenticatedDataSourceSet
+import com.wire.kalium.logic.data.user.UserId
+
+
+interface AuthenticatedDataSourceSetProvider {
+    fun get(userId: UserId): AuthenticatedDataSourceSet?
+    fun add(userId: UserId, authenticatedDataSourceSet: AuthenticatedDataSourceSet)
+    fun delete(userId: UserId)
+}
+
+internal object AuthenticatedDataSourceSetProviderImpl : AuthenticatedDataSourceSetProvider {
```

Comment:
We could quickly write test for this guy.

Reply:
Some classes needed to create `AuthenticatedDataSourceSet` have no interface, so the test will not be quickly. I will add a TODO and raise a new PR with the test.